crates/wash-cli/src/run.rs

```diff
@@ -0,0 +1,266 @@
+    pub allow_connect_udp: Vec<String>,
+
+    /// Allow the component to bind to any interface, port and protocol.
+    #[clap(long = "allow-net-bind-all")]
+    pub allow_bind_all: bool,
+
+    /// Allow the component to bind to a specific interface, specific port ranges but any protocol.
+    ///
+    /// The format expected is the same as expected of `--allow-net-connect`.
+    #[clap(long = "allow-net-bind")]
+    pub allow_bind: Vec<String>,
+
+    /// Allow the component to bind to any interface and port using the TCP protocol.
+    #[clap(long = "allow-net-bind-tcp-all")]
+    pub allow_bind_tcp_all: bool,
+
+    /// Allow the component to bind to a specific interface and specific port ranges using the TCP protocol.
+    ///
+    /// The format expected is the same as expected of `--allow-net-connect`.
+    #[clap(long = "allow-net-bind-tcp")]
+    pub allow_bind_tcp: Vec<String>,
+
+    /// Allow the component to bind to any interface and port using the UDP protocol.
+    #[clap(long = "allow-net-bind-udp-all")]
+    pub allow_bind_udp_all: bool,
+
+    /// Allow the component to bind to a specific interface and specific port ranges using the UDP protocol.
+    ///
+    /// The format expected is the same as expected of `--allow-net-connect`.
+    #[clap(long = "allow-net-bind-udp")]
+    pub allow_bind_udp: Vec<String>,
+}
+
+struct ParsedDir {
+    path: PathBuf,
+    file_perms: Option<FilePerms>,
+    guest_dir: Option<String>,
+}
+
+enum FilePerms {
+    ReadOnly,
+    ReadWrite,
+}
+
+impl FromStr for FilePerms {
+    type Err = anyhow::Error;
+
+    fn from_str(s: &str) -> std::result::Result<Self, Self::Err> {
+        match s {
+            "ro" => Ok(Self::ReadOnly),
+            "rw" => Ok(Self::ReadWrite),
+            _ => Err(anyhow!(
+                "file permission is incorrect, 'ro' or 'rw' expected"
+            )),
+        }
+    }
+}
+
+impl Into<WasmtimeFilePerms> for FilePerms {
+    fn into(self) -> WasmtimeFilePerms {
+        match self {
+            FilePerms::ReadOnly => WasmtimeFilePerms::READ,
+            FilePerms::ReadWrite => WasmtimeFilePerms::WRITE,
+        }
+    }
+}
+
+fn parse_dir(raw: impl AsRef<str>) -> Result<ParsedDir> {
+    let mut parts = raw.as_ref().split(':');
+
+    let raw_path = parts
+        .next()
+        .ok_or(anyhow!("dir flags require at least a path"))?;
+
+    Ok(ParsedDir {
+        path: PathBuf::from(raw_path),
+        file_perms: parts
+            .next()
+            .map(|raw| FilePerms::from_str(raw))
+            .transpose()?,
+        guest_dir: parts.next().map(|raw| raw.to_owned()),
+    })
+}
+
+pub async fn handle_command(cmd: RunCommand, _output_kind: OutputKind) -> Result<CommandOutput> {
+    let runtime = LocalRuntime::new()?;
+    let mut ctx_builder = CtxBuilder::new();
+
+    ctx_builder.set_reference(cmd.reference.clone());
+    ctx_builder.wasi_ctx().args(&cmd.args);
+
+    ctx_builder.wasi_ctx().inherit_stdio();
```

Comment:
❓ Should we give control over the stdin, stderr and stdout?

I don't see any reason to block those fd but maybe some of you have idea?
Maybe the most immediate reason is to not let components return https://en.wikipedia.org/wiki/ANSI_escape_code to avoid it controlling the user terminal?